### DISCUSSION TITLE
main.py runs in thonny, but not when powered up independantly

### DISCUSSION THREAD

**Discussion by Zimajays** [micropython/micropython#15121] [RP2040 / Pico]
Hello all

I am fairly new to micropython, but have successfully done a few things.  I am trying to get my main.py to run when powered up with out connecting to thonny.   It just does nothing.   

here is the code

import utime
from machine import Pin, I2C
from PicoDHT22 import PicoDHT22
import freesans20
import writer
from ssd1306 import SSD1306_I2C

i2c = I2C(0, scl=Pin(1), sda=Pin(0))
oled = SSD1306_I2C(128, 32, i2c)

dht_sensor=PicoDHT22(Pin(14,Pin.IN,Pin.PULL_UP),dht11=True)

relaypin = Pin(10, mode = Pin.OUT, value = 0)

while True:
    T,H = dht_sensor.read()
    utime.sleep_ms(1)
    
    font_writer = writer.Writer(oled, freesans20)
    font_writer.set_textpos(10, 10)
    font_writer.printstring(str("{:.1f}C".format(T)))
    font_writer.set_textpos(70, 10)
    font_writer.printstring(str("{:.1f}%".format(H)))
    oled.show() 
    utime.sleep(2)
    
    if T > 27:
        relaypin (0)
    else:
        relaypin (1)

Like I said works fine when connected to thonny, but when I restart it nothing happens.  What am I missing?

**Comment by robert-hh** [comment]
Probably main.py was never copied to the board's file system. That's a common pitfall when using Thonny. Try copying main.py to the device. You may use Thonny or mpremote.py for that purpose.

**Reply by Zimajays** [reply]
thanks for your reply.  I checked the os and its there as main.py

**Comment by Zimajays** [comment]
![Screenshot 2024-05-25 174107](https://github.com/micropython/micropython/assets/132131664/24fdf041-6070-47fc-be6c-659730f4c33a)

**Reply by robert-hh** [reply]
Are the other files imported by main.py in the lib directory, and does sys.path include lib?

**Reply by Zimajays** [reply]
The main imports all files from the lib.  I did have it set up so there was no lib and they were in the same place as main.py but the result was still the same.  Not sure on sys.path could you explain that to me please.

**Comment by Zimajays** [comment]
![Screenshot 2024-05-25 182805](https://github.com/micropython/micropython/assets/132131664/355d805a-5d8a-43ae-b2fd-b6269c4e9a72)

**Reply by robert-hh** [reply]
You can try tu use a simple terminal emulator like Putty or screen or mpremote and see if   errors pop up at boot or soft reset using ctrl-d.

**Reply by Zimajays** [reply]
yeh tried putty and coolterm and nothing.  even add some print commands at each step and runs fine on thonny but nothing prints on putty or coolterm.   ctrl d works once, but then when I unplug the power source and plug it back in - nothing.  Tried different uf2 versions.  I am at the point of uninstalling thonny and reinstalling thonny to see if that makes a difference

**Reply by robert-hh** [reply]
So you get a REPL prompt with Putty when you push Ctrl-C? That should stop your code unless Ctrl-C interrupt is disabled. With PuTTY you have to disable flow control. Do you use the right USB port?

If you unplug power while Putty is connected it will not re-establish the Serial connection. Then, you have to re-start PuTTY.
P.S.:. Did you shut down Thonny while trying Putty?

**Comment by Zimajays** [comment]
Ok so went back to basics and it would appear that its the font writer part of my code.   I wrote basic codes for each hardware, DHT relay and oled.  And all work as main expect the oled font writer one.

**Comment by Zimajays** [comment]
Just as an update - it was the oled - I tried a different oled and the code works fine.  Many thanks @robert-hh for taking the time to reply.
